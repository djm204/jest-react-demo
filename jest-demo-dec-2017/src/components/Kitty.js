import React from 'react';

export const Kitty = (props) => {
    const {kittyImg} = props;
    return (
        <div className="kitty">
            <img src={kittyImg} alt="Das Kitty"/>
        </div>
    );
}

export const AddKittyButton = (props) => {
    return (
        <button className={props.class} onClick={props.onClickCallback}>
            {props.children}
        </button>
    );
}

export const AddKittyActions = (props) => (
    <div>
        <AddKittyButton class="AddKitty" onClickCallback={props.AddKitty}> Add Kitty </AddKittyButton>
        <AddKittyButton class="RemoveKitty" onClickCallback={props.RemoveKitty}> Remove Kitty </AddKittyButton>
    </div>
)