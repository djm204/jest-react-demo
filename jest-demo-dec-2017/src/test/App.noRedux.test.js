
import React from 'react';
import ReactDOM from 'react-dom';
import _ from 'lodash';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import App from '../App.noRedux';
import {AddKittyButton, AddKittyActions} from '../components/Kitty'


Enzyme.configure({ adapter: new Adapter() });


it('renders without crashing', () => {
  const appRender = renderer.create(<App />);
  expect(appRender.toJSON()).toMatchSnapshot();
});

it('Call back functions are firing properly.', () => {
    const addKittyBtn = renderer.create(<AddKittyButton onClickCallback={e => 1} />);
    const addKittyBtnResult = addKittyBtn.toJSON().props.onClick();
    expect(addKittyBtnResult).toEqual(1);

    //Enzyme to test children
    const kittyActionsWrapper = Enzyme.mount(<AddKittyActions AddKitty={e => 1} RemoveKitty={e => -1} />);
    expect(kittyActionsWrapper.find('.AddKitty').props().onClick()).toEqual(1);
    expect(kittyActionsWrapper.find('.RemoveKitty').props().onClick()).toEqual(-1);
});

