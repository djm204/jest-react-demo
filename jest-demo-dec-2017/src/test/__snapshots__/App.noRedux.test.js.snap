// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="kitty.png"
    />
    <h1
      className="App-title"
    >
      Add Kittehs
    </h1>
    <div>
      <button
        className="AddKitty"
        onClick={[Function]}
      >
         Add Kitty 
      </button>
      <button
        className="RemoveKitty"
        onClick={[Function]}
      >
         Remove Kitty 
      </button>
    </div>
  </header>
  <div
    className="addKittiesForm"
  />
</div>
`;
